### ISSUE TITLE
WebRTC from service worker

### ISSUE THREAD

**Issue by backkem** [ipfs/ipfs-service-worker#13]
It seems there is a new window of opportunity for making WebRTC work from the service worker, see webtorrent/community#11. Are you guys interested in/tracking/weighing in on this discussion?

**Comment by lidel** [comment]
Yes, we've been signaling the need to browser vendors for some time.
Others in community been asking for this since  [2015](https://lists.w3.org/Archives/Public/public-webrtc/2015May/0157.html) and AFAIK current discussion on w3c lists highlights various use cases, including data channels, which IPFS would love to use in Service Worker.

IIUC we are interested in "1. Enable data channel support in workers":
- https://lists.w3.org/Archives/Public/public-webrtc/2018Jun/0176.html 

On IPFS side there is a meta-discussion about transports in browsers at: https://github.com/ipfs/js-ipfs/issues/1088

**Comment by Tryptophan** [comment]
Would it be possible to use this polyfill? https://github.com/johanholmerin/worker-webrtc Seems to at least support datachannels.
